From 6719bf15ade846c8542b66ff58d264991e9a12eb Mon Sep 17 00:00:00 2001
From: Marcelo Leitner <mleitner@redhat.com>
Date: Tue, 21 Oct 2014 16:21:30 -0400
Subject: [net] netfilter: ulog: compat with new structure

Message-id: <20141021162129.22888.39023.stgit@localhost.localdomain>
Patchwork-id: 98021
O-Subject: [RHEL7 net PATCH 124/124] redhat: netfilter: ULOG: compat with new structure
Bugzilla: 1148041 1155088
RH-Acked-by: Florian Westphal <fwestpha@redhat.com>

RHEL specific.

ULOG was removed from upstream, but we cannot remove it from RHEL. So,
as described in https://bugzilla.redhat.com/show_bug.cgi?id=1148041#c5 ,
we have to make ULOG compatible with the new structure.

This commit makes ULOG have the same type as NFLOG and thus will
conflict with each other.

Unfortunatelly, nf_log_register, doesn't really check for this conflict.
It's been posted upstream (#c6), but not integrated yet.

Signed-off-by: Jiri Benc <jbenc@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/net/ipv4/netfilter/ipt_ULOG.c b/net/ipv4/netfilter/ipt_ULOG.c
index 32b0e97..de23bae 100644
--- a/net/ipv4/netfilter/ipt_ULOG.c
+++ b/net/ipv4/netfilter/ipt_ULOG.c
@@ -394,6 +394,7 @@ static struct xt_target ulog_tg_reg __read_mostly = {
 
 static struct nf_logger ipt_ulog_logger __read_mostly = {
  .name  = "ipt_ULOG",
+ .type  = NF_LOG_TYPE_ULOG,
  .logfn  = ipt_logfn,
  .me  = THIS_MODULE,
 };
@@ -473,8 +474,11 @@ static int __init ulog_tg_init(void)
  if (ret < 0)
   goto out_target;
 
- if (nflog)
-  nf_log_register(NFPROTO_IPV4, &ipt_ulog_logger);
+ if (nflog) {
+  ret = nf_log_register(NFPROTO_IPV4, &ipt_ulog_logger);
+  if (ret < 0)
+   return ret;
+ }
 
  return 0;
 
-- 
1.7.1